src/run/level.rs

```diff
@@ -53,386 +80,118 @@ pub struct LevelCompletedEvent;
 /// Struct to manage a level
 #[derive(Clone, Debug)]
 pub struct Level {
-    /// Index of the current phase
-    timeline_idx: usize,
-    /// Timeline
-    pub timeline: LevelTimeline,
-    /// Tracks time of phases
-    pub phase_timer: Option<Timer>,
-    /// Tracks time between spawns
-    pub spawn_timer: Option<Timer>,
+    pub completed_phases: VecDeque<LevelPhase>,
```

Comment:
Maybe `pub type LevelPhases = VecDeque<LevelPhase>` to save some typing and ease changing the implementation (atleast as much as others parts of the code base rely on common traits/interfaces)